llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp

```diff
@@ -257,6 +257,20 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
       .clampNumElements(0, v2s64, v2s64)
       .moreElementsToNextPow2(0);
 
+  getActionDefinitionsBuilder(G_FABS)
+      .legalFor({MinFPScalar, s32, s64, v2s32, v4s32, v2s64})
+      .legalIf([=](const LegalityQuery &Query) {
+        const auto &Ty = Query.Types[0];
```

Comment:
no auto, and no reference. this is just LLT. You can also express this as a conditional legalFor, or swap out the legalFor list based on the HasFP16 feature

Reply:
This was actually copied from above

Reply:
Yes, but that could also be better. These predicates can be called many times so they ideally should be as simple as possible to get to the legal case